nscplugin/src/main/scala/scala/scalanative/nscplugin/NirNameEncoding.scala

```diff
@@ -55,14 +64,19 @@ trait NirNameEncoding { self: NirCodeGen =>
 
     if (sym == String_+) {
       genMethodName(StringConcatMethod)
-    } else if (isExternModule(owner)) {
-      ownerId member id tag "extern"
+    } else if (isExternModule(sym.owner)) {
+      val name = sym.annotations.collectFirst {
+        case ann if ann.symbol == NameClass =>
+          val Apply(_, List(Literal(Constant(name: String)))) = ann.tree
+          name
+      }.getOrElse(id)
```

Comment:
This is copy-pasted. You should factor it out in a `def nativeNameOf(sym)` or something like that. You might also want to have a look at the implementation of the corresponding feature in Scala.js at https://github.com/scala-js/scala-js/blob/608baf718e4366dfdcb0745f13facb2b438344f6/compiler/src/main/scala/org/scalajs/core/compiler/JSGlobalAddons.scala#L133-L139
Note that it reuses more scalac things (like `getAnnotation` and `stringArg`), and also it properly deals with setters.